backend/app.py

```diff
@@ -15,28 +16,53 @@
 # Initialize the database
 db = SQLAlchemy(app)
 
+# Initialize Flask-Migrate
+migrate = Migrate(app, db)  # Add this line
+
+# Association table for many-to-many relationship between Game and Player
+match_players = db.Table(
+    'match_players',
+    db.Column('match_id', db.String, db.ForeignKey('game.match_id'), primary_key=True),
+    db.Column('player_id', db.String, db.ForeignKey('player.player_id'), primary_key=True)
+)
+
 # Define a table to store games
 class Game(db.Model):
     id = db.Column(db.Integer, primary_key=True)
-    match_id = db.Column(db.String, unique=True, nullable=False)  # Match ID
-    date = db.Column(db.String, nullable=False)  # Date of the match
-    queue_type = db.Column(db.String, nullable=False)  # Queue type
-    player_count = db.Column(db.Integer, nullable=False)  # Total players
-    human_count = db.Column(db.Integer, nullable=False)  # Human players
-    game_length = db.Column(db.Integer, nullable=False)  # Duration in seconds
+    match_id = db.Column(db.String, unique=True, nullable=False)
+    date = db.Column(db.String, nullable=False)
+    queue_type = db.Column(db.String, nullable=False)
+    player_count = db.Column(db.Integer, nullable=False)
+    human_count = db.Column(db.Integer, nullable=False)
+    game_length = db.Column(db.Integer, nullable=False)
+
+    # Many-to-Many Relationship
+    players = db.relationship('Player', secondary=match_players, backref='matches')
 
     def __repr__(self):
         return f"<Game {self.match_id}>"
 
+# Define a table to store players
+class Player(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    player_id = db.Column(db.String, unique=True, nullable=False)
+    player_name = db.Column(db.String, nullable=False)
+    player_slot = db.Column(db.Integer, nullable=False)
+    legion = db.Column(db.String, nullable=False)
+    game_result = db.Column(db.String, nullable=False)
+    overall_elo = db.Column(db.Float, nullable=False)
+    classic_elo = db.Column(db.Float, nullable=False)
+    party_size = db.Column(db.Integer, nullable=False)
+    eco = db.Column(db.Float, nullable=False)
+    legion_elo = db.Column(db.Float, nullable=False)
```

Comment:
I would put these in the match players table since I think they are specific to a single match. (Feel free to ignore if I am incorrect in that assumption).

Reply:
After some research, it looks like keeping them in the Player model to streamline the data structure and simplify queries is probably correct given that each player is uniquely identified within a match through match_id.